build.rs

```diff
@@ -8,10 +8,41 @@ fn main() {
         println!("cargo:rustc-cfg=feature=\"nightly\"");
     }
 
+    #[cfg(feature = "cuda")]
+    cuda::build_ptx();
+
     #[cfg(feature = "intel-mkl")]
     intel_mkl::link().unwrap();
 }
 
+#[cfg(feature = "cuda")]
+mod cuda {
+    pub fn build_ptx() {
+        // TODO build ptx file in source tree and don't call nvcc if so
```

Comment:
I think this can be done later, I'm not even sure if it's necessary. Once we have all the kernels in place we can see, but no need to complicate something that's pretty simple atm